services/venice-controller/src/main/java/com/linkedin/venice/controller/multitaskscheduler/StoreMigrationTask.java

```diff
@@ -592,10 +601,60 @@ private void rescheduleStatusVerification() {
   }
 
   private List<String> findNotReadyFabrics() {
+    if (fabricReadyMap.isEmpty()) {
+      throw new IllegalArgumentException("fabricReadyMap must not be empty for non-single datacenter setup.");
+    }
     return fabricReadyMap.entrySet()
         .stream()
         .filter(e -> Boolean.FALSE.equals(e.getValue())) // keep entries with false
         .map(Map.Entry::getKey) // take the key
         .collect(Collectors.toList());
   }
+
+  private boolean updateSrcStoreReadWriteStatus(boolean enableReads, boolean enableWrites) {
+    if (srcStoreReadWriteEnabled == (enableReads && enableWrites)) {
+      // No need to update
+      return true;
+    }
+    ControllerResponse updateStoreControllerResponse = srcControllerClient.updateStore(
+        record.getStoreName(),
+        new UpdateStoreQueryParams().setEnableReads(enableReads).setEnableWrites(enableWrites));
+
+    if (updateStoreControllerResponse.isError()) {
+      LOGGER.error(
+          "Failed to {} reads and {} writes for store {} in original cluster {} on step {} due to error: {}",
+          enableReads ? "enable" : "disable",
+          enableWrites ? "enable" : "disable",
+          record.getStoreName(),
+          record.getSourceCluster(),
+          record.getCurrentStepEnum(),
+          updateStoreControllerResponse.getError());
+      return false;
+    }
+    boolean updated = false;
+    int maxRetries = 10;
+    for (int i = 0; i < maxRetries; i++) {
+      StoreInfo srcStoreInfo = srcControllerClient.getStore(record.getStoreName()).getStore();
+      if (srcStoreInfo.isEnableStoreReads() == enableReads && srcStoreInfo.isEnableStoreWrites() == enableWrites) {
+        updated = true;
+        srcStoreReadWriteEnabled = enableReads && enableWrites;
```

Comment:
Interesting; I am not sure how to interpret boolean usage. Could you please elaborate?